File: day9/power_of_four.cpp
/*
code by : vivek pandey

*/




class Solution {
public:
    bool isPowerOfFour(long long int num) {
     

/// bool ispowerofpower
	    bool ans ;
        if(num==0){
            ans = false;
        }
      else  if((num & (num-1))==0){
	        int cnt =0;
	    while(num){
		    if((num & 1)==0){
			cnt++;
		    }
		    num >>= 1;
	    }
	        if(cnt%2==0){
		    ans = true;
	        }
            
	        else{
		    ans = false;
	        }
 
        }
        
        else{
	    ans = false;
            }
        return ans;   
    }
};

File: day9/powerOfFour.java
/*
        **Auther Akash **
        **Approch***
        if a number is power of four then this number is never going to return non Zero reminder after devide.so after using % oprator for checking reminder and reduce the number by deviding by four     :)#happy Coding


*/


public class Solution{
public boolean isPowerOfFour(int num) {
    while(num>0){
        if(num==1){
            return true;
        }
 
        if(num%4!=0){
            return false;
        }else{
            num=num/4;
        }
    }
 
    return false;
}
    }
